moped-editor/src/views/projects/projectsListView/components/saveViewForm.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import FormControl from "@mui/material/FormControl";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Button, Grid } from "@mui/material";
+import { CheckCircle } from "@mui/icons-material";
+
+import * as yup from "yup";
+import ControlledTextInput from "src/components/forms/ControlledTextInput";
+
+const validationSchema = yup.object().shape({
+  description: yup.string().required(),
+});
+
+const SaveViewForm = ({ onSave, description }) => {
+  const {
+    handleSubmit,
+    control,
+    watch,
+    formState: { errors },
+  } = useForm({
+    defaultValues: { description: description },
+    resolver: yupResolver(validationSchema),
+  });
+
+  const areFormErrors = Object.keys(errors).length > 0;
+
+  return (
+    <form onSubmit={handleSubmit(() => onSave())}>
+      <Grid container spacing={2}>
+        <Grid item xs={12}>
+          <FormControl fullWidth>
+            <ControlledTextInput
+              fullWidth
+              label="Description"
+              name="description"
+              size="small"
+              control={control}
+              helperText="Required"
+            />
+          </FormControl>
+        </Grid>
+      </Grid>
+      <Grid container spacing={4} display="flex" justifyContent="flex-end">
+        <Grid item style={{ margin: 5 }}>
+          <Button
+            variant="contained"
+            color="primary"
+            startIcon={<CheckCircle />}
+            type="submit"
+            disabled={!watch("description")?.length || areFormErrors}
```

Comment:
Nice to catch if the description is empty! 🙏 The other day I learned about [Yup trim()](https://github.com/jquense/yup?tab=readme-ov-file#stringtrimmessage-string--function-schema) which can be chained in the validation and will also prevent empty spaces from saving in the form.

could you update that and also pass the mutation loading state into this component so submit button is only disabled when submitting? I'm drawing from [Chia's feedback](https://github.com/cityofaustin/atd-moped/pull/1557#pullrequestreview-2675722604) about keeping the submit button interactive to surface error messages.